kartothek/io/dask/delayed.py

```diff
@@ -41,15 +41,13 @@
 from ._utils import _maybe_get_categoricals_from_index, map_delayed
 
 
-@delayed
 def _delete_all_additional_metadata(dataset_factory):
     delete_indices(dataset_factory=dataset_factory)
     delete_common_metadata(dataset_factory=dataset_factory)
     # This is just a dummy return to chain
     return dataset_factory.dataset_uuid
 
 
-@delayed
```

Comment:
Out of curiosity: Did you change this for style or is there a deeper reason?

Reply:
It has a deeper reason: pickle cannot find the function if it is a delayed object. So this works:

```python
def f():
    ...
```

but this

```python
@delayed
def f():
    ...
```

compiles to

```python
def f():
    ...

f = delayed(f)
```

and therefore pickle cannot find `f` (the function) anymore, because it is now a delayed object:

```
_pickle.PicklingError: Can't pickle <function f at 0x7f56267fc840>: it's not the same object as my_module.f
```

 That's pretty annoying though.

Reply:
Good to know. I always considered the decorators as a very elegant way to write these functions.